docs/userguide/interfaces.rst

```diff
@@ -50,6 +50,16 @@ reserve the right of speeding up the deprecation cycle and shortening deprecatio
 Note that these are exceptional circumstances and that the project will
 carefully attempt to find alternatives before resorting to unscheduled removals.
 
+.. important::
+   In the context of ``setuptools``, the introduction of :py:mod:`warnings`
+   (including deprecation warnings) is not considered a breaking change *per se*.
+   Instead it is considered a backwards compatible *communication action* that
+   precedes an upcoming breaking change. This is becauset code
+   containing warnings typically does not fail and can successfully terminate
+   execution, unless users explicitly opt into transforming those warnings
+   into errors (e.g., via Python's :external+python:ref:`-W option or
```

Comment:
Please realize, that such settings sometimes aren't under people control.

Default sphinx invocation in the RTFD is something like this:
```
python -m sphinx -T -W --keep-going -b html -d _build/doctrees -D language=en . $READTHEDOCS_OUTPUT/html 
```

People can't just "opt out" here.

Reply:
Thank you very much for the review.

Could you please clarify this comment?

According to https://docs.readthedocs.com/platform/stable/config-file/v2.html#sphinx-fail-on-warning the default behaviour is not to fail on warnings (the docs mention `sphinx.fail_on_warning` being `false` by default).

Reply:
Sorry, my fault.  Default value here is false. (sent from phone)